machine-learning/tutorials/TransferLearningTF/Program.cs

```diff
@@ -0,0 +1,182 @@
+        static readonly string _trainImagesFolder = Path.Combine(_assetsPath, "inputs-train", "data");
+        static readonly string _predictImagesFolder = Path.Combine(_assetsPath, "inputs-predict", "data");
+        static readonly string _inceptionPb = Path.Combine(_assetsPath, "inputs-train", "inception", "tensorflow_inception_graph.pb");
+        static readonly string _inputImageClassifierZip = Path.Combine(_assetsPath, "inputs-predict", "imageClassifier.zip");
+        static readonly string _outputImageClassifierZip = Path.Combine(_assetsPath, "outputs", "imageClassifier.zip");
+        private static string LabelTokey = nameof(LabelTokey);
+        private static string ImageReal = nameof(ImageReal);
+        private static string PredictedLabelValue = nameof(PredictedLabelValue);
+        // </SnippetDeclareGlobalVariables>
+
+        static void Main(string[] args)
+        {
+            // Create MLContext to be shared across the model creation workflow objects 
+            // <SnippetCreateMLContext>
+            MLContext mlContext = new MLContext(seed:1);
+            // </SnippetCreateMLContext>
+
+            // <SnippetCallBuildAndTrain>
+            BuildAndTrainModel(mlContext, _trainTagsTsv, _trainImagesFolder, _inceptionPb, _outputImageClassifierZip);
+            // </SnippetCallBuildAndTrain>
+
+            // <SnippetCallClassifyImages>
+            ClassifyImages(mlContext, _predictTagsTsv, _predictImagesFolder, _outputImageClassifierZip);
+            // </SnippetCallClassifyImages>
+        }
+
+        // <SnippetImageNetSettings>
+        private struct ImageNetSettings
+        {
+            public const int ImageHeight = 224;
+            public const int ImageWidth = 224;
+            public const float Mean = 117;
+            public const float Scale = 1;
+            public const bool ChannelsLast = true;
+        }
+        // </SnippetImageNetSettings>
+
+        // Build and train model
+        public static void BuildAndTrainModel(MLContext mlContext, string dataLocation, string imagesFolder, string inputModelLocation, string outputModelLocation)
+        {
+
+            Console.WriteLine("Read model");
+            Console.WriteLine($"Model location: {inputModelLocation}");
+            Console.WriteLine($"Images folder: {_trainImagesFolder}");
+            Console.WriteLine($"Training file: {dataLocation}");
+            Console.WriteLine($"Default parameters: image size=({ImageNetSettings.ImageWidth},{ImageNetSettings.ImageHeight}), image mean: {ImageNetSettings.Mean}");
+
+            // <SnippetLoadData>
+            var data = mlContext.Data.ReadFromTextFile<ImageNetData>(path: dataLocation, hasHeader: true);
+            // </SnippetLoadData>
+
+            // <SnippetMapValueToKey1>
+            var estimator = mlContext.Transforms.Conversion.MapValueToKey(outputColumnName: LabelTokey, inputColumnName: DefaultColumnNames.Label)
```

Comment:
For this section, try to stay consistent with regards to column names. Some column names (i.e. ImageReal) are variables while others are string literals (i.e. "input").

Reply:
The reason the "softmax2..." and "input" are literals is those are coming from and to the inception TF model